### PR TITLE
switches github event api to search api and adds avatar

### PR DESCRIPTION
**Pull request by andrewjt71**
- Adds PR status
- Switches from using Github event api to Github search api
- Adds avatar

screen shot 2015-10-30 at 17 07 11

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -99,20 +92,20 @@ app.get('/', function(req, res) {
...
-    }
+    getUserEvent(req.query.username).then(function() {
+        var length,
+            statements;
 
-    q.all(promises).then(function() {
-        var length = octoberOpenPrs.length;
-        var statements = ["It's not too late to start!", "You can do it.", "Half way there.", "Almost there!", "Way to go!", "Now you're just showing off."];
+        length = octoberOpenPrs.length;
+        statements = ["It's not too late to start!", "You can do it.", "Half way there.", "Almost there!", "Way to go!", "Now you're just showing off."];
```
Comment (IrregularShed) [inline index.js]: Might actually be a bit late to start now :)

**Comment by IrregularShed** [comment]
:+1:

File: index.js (hunk 1 above)
Comment (jenkoian) [inline index.js]: :)

**Comment by jenkoian** [comment]
This all looks really good mate. Just testing it out locally and have come across a peculiar issue which means it probably can't be merged, but it's not your fault!

I use a personal access token to help with rate limiting the API. When this is used anyone within my organization (i.e. workplace) returns private as well as public repositories in the search results. So if you search for someone in the organization you could potentially see private PRs. 

I have raised an issues with Github to see if they can add a public/private filter parameter to their search API and will keep you updated. :+1:

File: index.js (hunk 1 above)
Comment (andrewjt71) [inline index.js]: lol

**Comment by andrewjt71** [comment]
:+1: Can you not use a new personal access token, to which you only grant access to private repositories? I may be missing the point though.

**Comment by jenkoian** [comment]
Yep you're right! You can also do something like is:public too it seems :+1:

**Comment by andrewjt71** [comment]
:+1: Right you are, all done!

File: index.js [hunk 2]
```diff
@@ -42,51 +42,44 @@ github.authenticate({
...
-    var options = {
-        user: username,
-        page: pageNumber
-    };
-    if (cache.get('ETag?' + username + '&' + pageNumber) !== null) {
-        options.headers = {
-            "If-None-Match": cache.get('ETag?' + username + '&' + pageNumber)
-        };
-    }
+function getUserEvent(username) {
```
Comment (jenkoian) [inline index.js]: This function name isn't quite accurate now, perhaps rename to `getPullRequests()` or something?

File: index.js [hunk 3]
```diff
@@ -42,51 +42,44 @@ github.authenticate({
...
+        options;
 
-    github.events.getFromUserPublic(options, function(err, res) {
+    deferred = q.defer();
+
+    options = {
+        q: 'created:2015-10-01..2015-10-31+type:pr+is:public+author:' + username
+    };
 
+    github.search.issues(options, function(err, res){
```
Comment (jenkoian) [inline index.js]: Minor but space before the curly brace :)

File: index.js [hunk 4]
```diff
@@ -42,51 +42,44 @@ github.authenticate({
...
-            openedPrs = _.filter(prs, {'payload': {action: "opened"}});
-        cache.put('PullRequest?' + username + '&' + pageNumber, openedPrs);
+        _.each(res.items, function(event){
+            var repo = event.pull_request.html_url.substring(0, event.pull_request.html_url.search('/pull'));
 
-        _.filter(openedPrs, function(event) {
-            return event.payload.pull_request.created_at.substr(0, 7) === '2015-10';
-        }).forEach(function(event) {
-            octoberOpenPrs.push(event);
+            if(userImage == null) {
```
Comment (jenkoian) [inline index.js]: Space after if

File: index.js [hunk 5]
```diff
@@ -42,51 +42,44 @@ github.authenticate({
...
+                title: event.title,
+                url: event.html_url,
+                state: event.state
+            }
+            octoberOpenPrs.push(returnedEvent);
         });
 
         deferred.resolve();
-    });
+    })
```
Comment (jenkoian) [inline index.js]: Keep the semi colon.

File: views/partials/prs.hbs [hunk 6]
```diff
@@ -1,4 +1,5 @@
 {{#exists prs}}
+  <img id="avatar" src="{{userImage}}"\>
```
Comment (jenkoian) [inline views/partials/prs.hbs]: should be />

File: views/partials/prs.hbs [hunk 7]
```diff
@@ -12,7 +13,7 @@
     <div class="large-12 columns">
       <ul class="pulls">
         {{#each prs}}
-          <li><a href="http://github.com/{{this.repo.name}}/pull/{{this.payload.number}}">{{this.repo.name}} - #{{this.payload.number}}</a></li>
+        <li><a href="{{this.url}}">{{this.repo_name}} - #{{this.title}} ({{this.state}})</a></li>
```
Comment (jenkoian) [inline views/partials/prs.hbs]: Keep the indent.

**Pull request merged**